### PR TITLE
Hidden doors

### PR DESCRIPTION
**Pull request by nichmack**

### CODE REVIEW

File: models/room.js [hunk 1]
```diff
@@ -217,10 +217,14 @@ RoomSchema.methods.look = function (socket, short) {
     output += `<span class="silver">${this.desc}</span>\n`;
   }
 
-  if (this.inventory && this.inventory.length > 0) {
+  if (this.inventory && this.inventory.filter(item => !item.hidden).length > 0) {
```
Comment (slashben) [inline models/room.js]: `if(Array.isArray(this.inventory)` is a better way to make sure the item is an array, if you are doing that, you can skip the length check.

File: models/room.js [hunk 2]
```diff
@@ -217,10 +217,14 @@ RoomSchema.methods.look = function (socket, short) {
...
     output += `<span class="darkcyan">You notice: ${this.inventory.filter(item => !item.hidden).map(item => item.displayName).join(', ')}.</span>\n`;
```
Comment (slashben) [inline models/room.js]: You don't want to filter the the inventory twice if you don't have to, if you end up taking the filter out of the if statement above this line, then this comment is moot.

File: models/room.js [hunk 3]
```diff
@@ -217,10 +217,14 @@ RoomSchema.methods.look = function (socket, short) {
...
   }
 
+  if (socket.user.admin && this.inventory && this.inventory.filter(item => item.hidden).length > 0) {
```
Comment (slashben) [inline models/room.js]: Double filter.

File: models/room.js [hunk 4]
```diff
@@ -231,10 +235,14 @@ RoomSchema.methods.look = function (socket, short) {
     output += `<span class="purple">Also here: <span class="teal">${displayNames}</span>.</span>\n`;
   }
 
-  if (this.exits.length > 0) {
+  if (this.exits.filter(exit => !exit.hidden).length > 0) {
```
Comment (slashben) [inline models/room.js]: Double filter.

File: models/room.js [hunk 5]
```diff
@@ -231,10 +235,14 @@ RoomSchema.methods.look = function (socket, short) {
...
     output += `<span class="green">Exits: ${this.exits.filter(exit => !exit.hidden).map(exit => Room.shortToLong(exit.dir)).join(', ')}</span>\n`;
   }
 
+  if (socket.user.admin && this.exits.filter(exit => exit.hidden).length > 0) {
```
Comment (slashben) [inline models/room.js]: double filter.

File: commands/move.js [hunk 6]
```diff
@@ -120,6 +120,12 @@ module.exports = {
       return;
     }
 
+    //general public cannot enter hidden rooms
+    if (exit.hidden && !socket.user.admin) {
```
Comment (slashben) [inline commands/move.js]: You can only enter a hidden room if you're an admin?

Comment (nichmack) [reply commands/move.js]: This is so players cannot (un)wittingly enter a room that is currently hidden from them.

**Pull request merged**